### ISSUE TITLE
Refactoring of UnitVector

### ISSUE THREAD

**Issue by mforets** [JuliaReach/LazySets.jl#616]
By the name `UnitVector` i would expect that the vector has norm one. But the constructor is more general: 

```julia
julia> LazySets.Approximations.UnitVector(1, 2, 2.0)
2-element LazySets.Approximations.UnitVector{Float64}:
 2.0
 0.0
```

Proposal: for many use cases we will need the default "1.0" entry, so that `LazySets.Approximations.UnitVector(1, 2)` corresponds to `LazySets.Approximations.UnitVector(1, 2, 1.0)`

On the other hand, maybe this type should be pulled out of the `Approximations` module.

**Comment by schillic** [comment]
You are right, we can use let the type return `one(N)` and `zero(N)` only. This has the benefit that the type does not have to store the element. If we need the more general behavior, we should add a new type.

> this type should be pulled out of the Approximations module

Yes, definitely.

**Comment by schillic** [comment]
About the name: A [unit vector](https://en.wikipedia.org/wiki/Unit_vector) just has length 1. There seems to be no specific name for vectors of the form `(0, ..., 0, 1, 0, ..., 0)`. The best I could come up with is `BinaryUnitVector`.

However, I changed my mind about having two types for these two vectors. The reason is that we have to implement methods two times (or introduce another interface).

And furthermore, I changed my mind about `UnitVector` completely. I am not sure how efficient that type actually is because we have almost no implementations for it. On average we might benefit much more from using established `SparseVector`s instead, which should be almost as efficient.
So I propose to replace `UnitVector` constructors by a helper method (e.g., `binary_unit_vector`) that creates a sparse vector instead. At some point we can just replace the implementation of that function by the constructor of a new `AbstractVector` type again.
